# Documentation

This wrapper produces local BLAST databases from nucleotide or protein FASTA files as input.

This wrapper takes fasta file. The database type depends on the output extension.
Use `.pdb` extension for protein database or use `.ndb` extension for nucleotide database

**Required input:**

- Fasta file.

**Required output:**

- multiple files with different extensions (e.g. .nin, .nsq, .nhr for nucleotides or .pin, .psq, .phr for proteins)
 
**Required parameters**

- **options**: a list of valid `makeblastdb` options.

**Log**:

- log file generated by makeblastdb


# Configuration

    ##############################################################################
    # Local blast database with makeblastdb
    #
    # :Parameters:
    #   
    # - options: any options recognised by makeblastdb
    makeblastdb:
        options: ''

# Example

    rule makeblastdb_nucleotide:
        input:
            "nice_genomes_list.fasta"
        output:
            "nice_genomes_list.fasta.ndb"
        params:
            options = '-title "Nice local db"'
        log:
            "nice_blast_logs.log"
        wrapper:
            "main/wrappers/blast/makeblastdb"


    rule makeblastdb_protein:
        input:
            "nice_protein_list.fasta"
        output:
            "nice_protein_list.fasta.pdb"
        log:
            "nice_blast_logs.log"
        wrapper:
            "main/wrappers/blast/makeblastdb"

# References

- https://www.ncbi.nlm.nih.gov/books/NBK569841/