# Documentation

This wrapper maps sequencing reads onto a reference using Minimap2

This wrapper takes single or paired data and map them on a reference.
The results file is a sorted and BAM file (not indexed).

**Required input:**

- **fastq**: the input FastQ files (single or paired)
- **reference**: the input genome used as a reference. No need for indexing

**Required output::**

- **sorted**: the sorted output BAM file

**Required parameters:**

- **options**: a list of valid fastqc options

**Log:**

- a log file generated by minimap2 is stored

# Configuration

    #############################################################################
    #  Bowtie2 read mapping
    #
    # :Parameters:
    #
    # - options: any options recognised by minimap2
    # - threads: number of threads to be used
    minimap2:
        options: ''
        threads: 4


# Example

    rule minimap2:
        input:
            fastq=["reads/{sample}.1.fastq", "reads/{sample}.2.fastq"],
            reference="genome.fasta"
        output:
            sorted="{sample}/{sample}.sorted.bam"
        log:
            "{sample}/{sample}.log"
        params:
            options=config['minimap2']['options']
        threads: 
            config['minimap2']['threads']
        wrapper:
            "main/wrappers/minimap2"

