# Documentation

Maps sequencing reads onto a reference using Bowtie2

This wrapper takes single or paired data and map them on a reference.
The results file is a sorted and indexed BAM file.

**Required input:**

- **fastq**: the input FastQ files (single or paired)

**Required output::**

- **bam** :the output BAM file.
- **sorted** :the sorted output BAM file. (optional but is computed)

Optional output (recommended):

- **sorted** If set, the BAM file will also be sorted and indexing provided
 
**Required parameters:**

- **options**: a list of valid fastqc options
- **index** the expected index prefix of the genome reference. For instance if the
  reference is in ref/hg38.fa, most  probably this parameter must be set to **ref/hg38**

**Log:**

- a log file generated by bowtie2 is stored

# Configuration

    #############################################################################
    #  Bowtie2 read mapping
    #
    # :Parameters:
    #
    # - options: any options recognised by 'bowtie2 index' command
    # - threads: number of threads to be used
    bowtie2:
        options: ''
        threads: 4


# Example

    rule bowtie2:
        input:
            fastq=input_data
        output:
            bam="{sample}/bowtie2/{sample}.bam
            sorted="{sample}/bowtie2/{sample}.sorted.bam
        params:
            options = config['bowtie2'][options'],
            index="reference/mygenome"
        threads:
            config['bowtie2"]["threads"]
        log:
            "{sample}/bowtie2/{sample}.log"
        wrapper:
            "main/wrappers/bowtie2/align"

