# Documentation

This wrapper calls FastQC on input data sets (paired or not)

This wrapper takes 1 or 2 FastQ files as input. It creates FastQC reports (HTML and zip file).
A **fastqc.done** file is generated once done. 

**Required input:**

- one FastQ file or two FastQ files (if paired data). Could also be a BAM file.

**Required output:**

- **done**: a filename used as a trigger of job done. Be aware that the root
  directory of this file is used to store the results. 

**Required parameters:**

- **options**: a list of valid FastQC options
- **working_directory**: place where results of FastQC are saved

**Log:**

- a log file generated by FastQC is stored

# Configuration

    ##############################################################################
    # FastQC section
    #
    # :Parameters:
    #
    # - options: string with any valid FastQC options
    fastqc:
        options: ''
        threads: 4

# Example

    rule fastqc:
        input:
            "{sample}_R1_.fastq", "{sample}_R2_.fastq"
        output:
            done = "{sample}/fastqc/fastqc.done"
        params:
            options = config['fastqc'][options'],
            working_directory = "{sample}/fastqc"
        threads:
            config['fastqc']["threads"]
        log:
            "{sample}/fastqc/fastqc.log"
        wrapper:
            "main/wrappers/fastqc"

# References

- http://www.bioinformatics.babraham.ac.uk/projects/fastqc/
