# Documentation

This wrapper takes an alignment of reads or fragments as input (BAM file) and generates a coverage track (bigWig or bedGraph) as output using **bamCoverage** from **deeptools**.

**Required input:**

- **input**: aligned reads in bam file.

**Required output::**

- **output**: coverage track in bigwig or bedGraph.
 
**Required parameters:**

- **options**: a list of valid bamCoverage options.

**Log:**

- a log file generated by bamCoverage is created.

**Note:**
The wrapper will generate index for the input bam file if it's missing.

# Configuration


    ##############################################################################
    # Deeptools bamCoverage
    #
    # :Parameters:
    #
    # - options: any options recognised by bamCoverage cli.
    # - threads: number of threads to be used.
    #
    bam_coverage:
        options: "--normalizeUsing RPGC --effectiveGenomeSize 2913022398"
        threads: 4


# Example

"""
    rule bam_coverage:
        input:
            "data/aligned.bam"
        output:
            "bam_coverage/align.bw"
        params:
            options = config["bam_coverage"]["options"]
        log:
            "{sample}/logs/bam_coverage.log"
        threads:
            config["bam_coverage"]["threads"]
        wrapper:
            "main/wrappers/deeptools/bam_coverage"
"""
