# Documentation

This wrapper maps sequencing reads onto a reference using **bwa**

This wrapper takes single or paired data and map them on a reference.
The results file is a sorted and indexed BAM file.

**Required input:**

- **fastq**: the input FastQ files (single or paired)

**Required output::**

- **bam** :the output BAM file.
- **sorted** :the sorted output BAM file. (optional but is computed)

Optional output (recommended):

- **sorted** If set, the BAM file will also be sorted and indexing provided
 
**Required parameters:**

- **options**: a list of valid bwa options


**Log:**

- a log file generated by bwa is created

**Note**

Uses also sambamba to view  and sort the output BAM file.

# Configuration


    ##############################################################################
    # BWA -  Mapping
    #
    # :Parameters:
    #
    # - options: any options recognised by BWA MEM tool.
    # - threads: number of threads to be used.
    # - tmp_directory: the temporary directory for sambamba
    #
    bwa_align:
        options: -T 30 -M
        threads: 4
        tmp_directory: ./tmp


# Example


    rule bwa_align:
        input:
            fastq=["reads/{sample}.1.fastq", "reads/{sample}.2.fastq"],
            bwa_bwt="index/genome.fasta.bwt",
            fai="index/genome.fasta.fai",
            reference="index/genome.fasta"
        output:
            sorted="{sample}/{sample}.sorted.bam"
        log:
            "{sample}/{sample}.log"
        params:
            options=config["bwa_mapping]['options']
            tmp_directory=config["bwa_mapping"]["tmp_directory]
        threads:
            config["bwa_align"]["threads"]
        wrapper:
            "main/wrappers/bwa/align"

