# Documentation

This wrapper assesses an assembly using **Quast**.

**Required input:**

- **fastq**: the input fastq files.
- **assembly**: the input assembly fasta file.

**Required output::**

- **output**: Any empty filename. The path defines the quast output directory.
 
**Required parameters:**

- **preset**: preset for application (single, pacbio, nanopore). Ignored if fastq are paired-end.
- **reference**: reference genome fasta file.
- **annotation**: file with genomic feature coordinates in the reference (GFF, BED, NCBI or TXT).
- **options**: a list of valid **Quast** options.

**Log:**

- a log file generated by quast is created.

# Configuration


    ##############################################################################
    # Quast
    #
    # :Parameters:
    #
    # preset: preset for application (single, pacbio, nanopore). Ignored if fastq are paired-end.
    # reference: reference genome fasta file.
    # annotation: file with genomic feature coordinates in the reference (GFF, BED, NCBI or TXT).
    # options: a list of valid quast.py options.
    #
    quast:
        preset: pacbio
        reference: path_to/knowned_ref.fa
        annotation: path_to/knowed_ref.gff
        options: "--fungus"
        threads: 4


# Example

"""
    rule quast:
        input:
            fastq="data/raw.fastq.gz",
            assembly="assembly/contigs.fasta"
        output: "quast/quast.done"
        params:
            preset=config["quast"]["preset"],
            reference=config["quast"]["reference"],
            annotation=config["quast"]["annotation"],
            options=config["quast"]["options"],
        log:
            "{sample}/logs/quast.log"
        threads:
            config["quast"]["threads"]
        wrapper:
            "main/wrappers/quast"
"""
