# Documentation

This wrapper calls salmon to map reads on a genome.
output directory is based on the parent directory of the output.

**Required input:**

- **fastq**: list with one or two Fastq files. Must be compressed in gzipped format.
- **index**: a trigger file from salmon_index output

**Required output:**

- **quant**: the output of salmon.

Since there is only one output, we may name it differently.

**Required parameters:**

- **options**: a list of valid salmon options. default to ""
- **genome_dir** 

**Log:**

- a log file generated by salmon

# Configuration

    ##############################################################################
    # Salmon mapping section.
    #
    # :Parameters:
    #
    # - options: string with any valid Salmon options. -l A identifies strandness
    #     automatically
    #
    salmon_mapping:
        options: '-l A'
        threads: 4
        resources:
            mem: 4G


# Example

    rule salmon_index
        # this should create the salmon_index/salmon.done file

    rule salmon_mapping:
        input:
            fastq="test.fastq.gz",
            index="salmon_index/salmon.done"  # trigger file
        output:
            quant = "salmon_mapping/quant.sf"
        params:
            options= " ",
        threads: 2
        log:
            "salmon/salmon.log"
        wrapper:
            "main/wrappers/salmon_mapping"


# References

