# Documentation

De novo transcriptome quantification from Illumina RNA-seq data using Trinity.

**Required input:**

- **fasta**: Path to the transcriptome assembly from Trinity.
- **left**: List of paths to fastq files for R1.
- **right**: (optional) List of paths to fastq files for R2.

**Required output:**

- **outdir**: output directory.
- **transcripts**: Path to abundance file for transcripts.
- **genes**: Path to abundance file for genes.

**Parameters:**

- **lib_type**: Type of strand-specific library (RF, FR, F or R).
- **est_method**: The method to use for quantification: kallisto (default), salmon or RSEM.

**Log:**

- a log file generated by Trinity.

# Configuration

	######################################################################
	# Trinity quantification section
	#
	# :Parameters:
	#
	# - lib_type: string strand-specificity library type
	# - est_method: string quantification method
	trinity_quantify:
		lib_type: 'RF'
		est_method: 'salmon'
		threads: 4
	
# Examples

	rule quantify:
		input:
			fasta="trinity_out_dir/Trinity.fasta",
			left="{sample}/fastp/{sample}_R1_trimmed.fastq",
			right="{sample}/fastp/{sample}_R2_trimmed.fastq",
		output:
			outdir=directory("{sample}/kallisto"),
			transcripts="{sample}/kallisto/abundance.tsv",
			genes="{sample}/kallisto/abundance.tsv.genes",
		params:
			lib_type="RF",
			est_method="salmon"
		log:
			"logs/trinity/{sample}_quantify.log"
		threads: 2
		wrapper:
			"main/wrappers/trinity_quantify"

# References

- https://github.com/trinityrnaseq/trinityrnaseq/wiki

