# Documentation

This wrapper builds a genome index with Bowtie2

This wrapper takes a reference and build its index. The output files are 
stored in a directory extracted from the genome reference name.

**Required input:**

- **reference**: the reference genome to index (FASTA format)

**Required output:**

- a list of expected output files. Need at least one. From the first file,
  the prefix is extracted and used by bowtie1 as **indexbase**.
 
**Required parameters**

- **options**: a list of valid fastqc options

**Log:**

- a log file generated by bowtie1 is stored

# Configuration

    #############################################################################
    #   bowtie1 build index
    #
    # :Parameters:
    #
    # - options: any options recognised by 'bowtie1-build' command
    # - threads: 
    bowtie1_build:
        options: ''
        threads: 4


# Example

    rule bowtie1_build:
        input:
            reference="genome.fa"
        output:
            "genome/bowtie1/genome.1.ebwt"
        params:
            options = config['bowtie1_build'][options'],
        threads:
            config['bowtie1_build']["threads"]            
        log:
            "logs/bowtie1_build/bowtie1_build.log"
        wrapper:
            "main/wrappers/bowtie1/build"1
