# Documentation

De novo transcriptome assembly from Illumina RNA-seq data using Trinity.

**Required input:**

- Path to fastq or fasta files (paired or single)

**Required output:**

- Path to output assembly fasta file

**Parameters:**

- **options**: a list of valid Trinity options.

**Ressources:**

- **mem_gb** or **mem_mb**: set the maximum memory usage.

**Log:**

- a log file generated by Trinity.

# Configuration

	######################################################################
	# Trinity section
	#
	# :Parameters:
	#
	# - options: string with any valid Trinity options
	trinity:
		options: ''
		threads: 4
		mem_gb: 10
	
# Examples

## Paired reads

	rule trinity:
		input:
			left = expand("{sample}_R1.fq.gz", sample=samples),
			right = expand("{sample}_R2.fq.gz", sample=samples)
		output:
			"trinity/trinity.fas"
		params:
			options = config["trinity"]["options"],
		threads:
			config["trinity"]["threads"]
		resources:
			mem_gb = config["trinity"]["mem_gb"]
		log:
			"trinity/trinity.log"
		wrapper:
			"main/wrappers/trinity"

## Single reads

	rule trinity:
		input:
			left = expand("{sample}.fq.gz", sample=samples)
		output:
			"trinity/trinity.fas"
		[...]

# References

- https://github.com/trinityrnaseq/trinityrnaseq/wiki
- https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/trinity.html
